### ISSUE TITLE
Alpine 3.16: unable to select packages: p7zip in arm/v6

### ISSUE THREAD

**Issue by ttionya** [jinfeihan57/p7zip#187]
Hello

Apline 3.16 `arm/v6` platform cannot find `p7zip`.

This is the log of my actions.

```shell
#0 0.403 fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/main/armhf/APKINDEX.tar.gz
#18 1.290 fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/community/armhf/APKINDEX.tar.gz
#18 2.234 ERROR: unable to select packages:
#18 2.323   p7zip (no such package):
#18 2.323     required by: world[p7zip]
```

I tried to download the apkindex file and found `p7zip@16.02-rc4` in alpine 3.15, but not in alpine 3.16.

https://dl-cdn.alpinelinux.org/alpine/v3.16/main/armhf/APKINDEX.tar.gz
https://dl-cdn.alpinelinux.org/alpine/v3.15/main/armhf/APKINDEX.tar.gz

**Comment by wdlkmpx** [comment]
Indeed, the alpine 3.16 armhf repository is missing p7zip

https://pkgs.alpinelinux.org/packages?name=p7zip&branch=v3.16&arch=armhf

p7zip in 3.15: https://pkgs.alpinelinux.org/packages?name=p7zip&branch=v3.15&arch=armhf

This is something that @ncopa should fix

**Comment by ttionya** [comment]
> From [this commit](https://git.alpinelinux.org/aports/commit/main/p7zip?id=147d4adbf083fe80024966207d8bb68438cea45c), package build for p7zip is disabled on armhf due to a failed test.

@antermin  Yes, I created the [issue](https://gitlab.alpinelinux.org/alpine/aports/-/issues/14008) in the alpinelinux repository. But I don't know when it will be fixed, or put on hold.

This issue should not be created here, I will close it later.

**Comment by jinfeihan57** [comment]
@ttionya Can zstd work on armhf?

**Comment by ttionya** [comment]
@jinfeihan57 ,

**Comment by ttionya** [comment]
@jinfeihan57 ,

Sorry I don't know, I just know that it didn't pass the zstd decompression test on `armhf`, pipeline [here](https://gitlab.alpinelinux.org/alpine/aports/-/jobs/767452#L1257).

**Comment by jinfeihan57** [comment]
Can you try [zstd v1.4.9](https://github.com/facebook/zstd/tree/v1.4.9) on your armhf device? I don't have armhf device.

**Comment by ttionya** [comment]
@jinfeihan57 ,

I don't have an `armhf` device either.

But I see that `zstd` was upgraded to v1.4.9 before the release of 17.04 wasn't it? So now the compiled program should be using `zstd` v1.4.9.

I cloned the master branch of this repository and compiled it on `armhf` Alpine 3.16 using GitHub Actions, the commands were obtained from the [alpine repository](https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/main/p7zip/APKBUILD) and it took 2 hours to compile. Here is my test repository [ttionya/p7zip-test](https://github.com/ttionya/p7zip-test), the wrong job is [here](https://github.com/ttionya/p7zip-test/runs/7501397385?check_suite_focus=true#step:5:1687), the excerpt of the error is as follows:

```txt
System ERROR:
E_FAIL
ERROR during : /app/check/../bin/7za a -t7z 7za433_7zip_zstd.7z 7za433_7zip_lzma -m0=zstd -mx=22
make: *** [makefile.common:101: test] Error 1
```
